backend/src/utils/validation.ts

```diff
@@ -9,3 +9,16 @@ export const zodDateOnlyStringSchema = z
         message: `String must be a valid date in ${DATE_ONLY_FORMAT} format`,
     })
     .transform((val) => coerceDateOnlyStringToDate(val));
+
+// validates file is string (base64) and meets the size limit
+export const zodBase64FileSchema = (
+    sizeInMb: number,
+): z.ZodEffects<z.ZodString, string, string> =>
+    z.string().refine((val) => {
+        return getOriginalByteSizeFromBase64(val) < sizeInMb * 1024 * 1024;
+    }, "File size too large");
+
+function getOriginalByteSizeFromBase64(base64Str: string) {
+    const numPaddingChars = (base64Str.match(/=/g) ?? []).length;
```

Comment:
I don't think this precision is required. I also am not sure the calculation is correct.

should be enough just to use `(base64Str.length * 3) / 4` alone

Reply:
`==`: last 4 characters decode to 1 byte, so compensate with -2
`=` : last 4 characters decode to 2 bytes, so compensate with -1
none: no padding, so no subtraction

you're right, it didn't need to be this complicated 😪  but will leave as is since it's already implemented